clearly-defined/src/main/kotlin/ClearlyDefinedService.kt

```diff
@@ -0,0 +1,160 @@
+
+import com.fasterxml.jackson.annotation.JsonCreator
+import com.fasterxml.jackson.annotation.JsonValue
+
+import retrofit2.Call
+import retrofit2.http.GET
+import retrofit2.http.Path
+import retrofit2.Retrofit
+import retrofit2.converter.jackson.JacksonConverterFactory
+
+import java.io.File
+import java.net.URL
+
+/**
+ * Interface for the ClearlyDefined REST API, based on code generated by https://app.quicktype.io/ from
+ * https://github.com/clearlydefined/service/tree/master/schemas.
+ */
+interface ClearlyDefinedService {
+    companion object {
+        fun create(server: Server): ClearlyDefinedService {
+            val retrofit = Retrofit.Builder()
+                .baseUrl(server.url)
+                .addConverterFactory(JacksonConverterFactory.create())
+                .build()
+
+            return retrofit.create(ClearlyDefinedService::class.java)
+        }
+    }
+
+    enum class Server(val url: String) {
+        PRODUCTION("https://api.clearlydefined.io"),
+        DEVELOPMENT("https://dev-api.clearlydefined.io"),
+        LOCALHOST("http://localhost:4000")
+    }
+
+    /**
+     * See https://github.com/clearlydefined/service/blob/master/schemas/curation-1.0.json.
+     */
+    data class Curation(
+        val described: Described? = null,
+        val files: List<FileEntry>? = null,
+        val licensed: Licensed? = null
+    )
+
+    data class Described(
+        val facets: Facets? = null,
+        val issueTracker: URL? = null,
+        val projectWebsite: URL? = null,
+        val releaseDate: String? = null,
+        val sourceLocation: SourceLocation? = null
+    )
+
+    data class Facets(
+        val data: List<String>? = null,
+        val dev: List<String>? = null,
+        val doc: List<String>? = null,
+        val examples: List<String>? = null,
+        val tests: List<String>? = null
+    )
+
+    data class SourceLocation(
+        val name: String,
+        val namespace: String? = null,
+        val path: String? = null,
+        val provider: Provider,
+        val revision: String,
+        val type: Type,
+        val url: String? = null
+    )
+
+    // See https://github.com/clearlydefined/service/blob/master/schemas/curation-1.0.json#L44
+    enum class Provider(val value: String) {
+        COCOAPODS("cocoapods"),
+        CRATES_IO("cratesio"),
+        DEBIAN("debian"),
+        GITHUB("github"),
+        MAVEN_CENTRAL("mavencentral"),
+        NPM_JS("npmjs"),
+        NUGET("nuget"),
+        PACKAGIST("packagist"),
+        PYPI("pypi"),
+        RUBYGEMS("rubygems");
+
+        companion object {
+            @JsonCreator
+            @JvmStatic
+            fun fromString(value: String) = enumValues<Provider>().single { value.equals(it.value, true) }
+        }
+
+        @JsonValue
+        override fun toString() = value
+    }
+
+    // See https://github.com/clearlydefined/service/blob/master/schemas/curation-1.0.json#L23.
+    enum class Type(val value: String) {
```

Comment:
You could make value a dynamic property:
`val value get() = name.toLowerCase().replace("_", "")`

Reply:
That wouldn't currently work for `DEBIAN_SOURCES("debsrc")`, although the name could of course be changed to `DEBIAN_SRC`, but I'm not sure about future values. What do you prefer, keep it as-is, or generalize with your getter and rename to `DEBIAN_SRC`? I'm a bit undecided because with your approach we deduce the value from the name, however it's the value that is defined in the spec, so if at all we should deduce the name from the value and not the other way around.

Reply:
It was just an idea, I'm fine with keeping it as is.